udaykalra/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,180 @@
 // limitations under the License.
 
 package com.google.sps;
-
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
+/**
+ * Returns a collection of TimeRange objects to indicate available meeting times
+ * for indicated people and duration given through a request object.
+ */
 public final class FindMeetingQuery {
+  public static final int OVERLAP_WITH_SAME_START = 1;
+  public static final int OVERLAP_WITH_SAME_END = 2;
+  public static final int OVERLAPS_END = 3;
+  public static final int OVERLAPS_START = 4;
+  public static final int OVERLAP_CONTAINS = 5;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    /* Create an output collection with a full day to cut(shorten or split)
+     * into the available ranges.
+     */
+    TimeRange fullDay = TimeRange.WHOLE_DAY;
+    Collection<TimeRange> availableRanges = new HashSet<>();
+    availableRanges.add(fullDay);
+
+    // Obtain data from the request.
+    Collection<String> attendees = request.getAttendees();
+    long requestDuration = request.getDuration();
+
+    Collection<TimeRange> busyRanges = new HashSet<>();
+
+    // Detect relevant busy time ranges based on attendees.
+    for (Event event : events) {
+      if (!(Collections.disjoint(attendees, event.getAttendees()))) {
```

Comment:
Nit: Parenthesis around (Collections....) are probably not needed.

Reply:
Done